plugins/filter/ipmath.py

```diff
@@ -0,0 +1,186 @@
+- debug:
+    msg: "{{ '2001::1' | ansible.netcommon.ipmath(10) }}"
+
+# Get the previous tenth address in IPv6
+- debug:
+    msg: "{{ '2001::5' | ansible.netcommon.ipmath(-10) }}"
+
+TASK [debug] **********************************************************************************************************
+ok: [localhost] => {
+    "msg": "192.168.1.10"
+}
+
+TASK [debug] **********************************************************************************************************
+ok: [localhost] => {
+    "msg": "192.168.0.251"
+}
+
+TASK [debug] **********************************************************************************************************
+ok: [localhost] => {
+    "msg": "192.168.1.6"
+}
+
+TASK [debug] **********************************************************************************************************
+ok: [localhost] => {
+    "msg": "192.168.1.1"
+}
+
+TASK [debug] **********************************************************************************************************
+ok: [localhost] => {
+    "msg": "192.168.1.252"
+}
+
+TASK [debug] **********************************************************************************************************
+ok: [localhost] => {
+    "msg": "2001::b"
+}
+
+TASK [debug] **********************************************************************************************************
+ok: [localhost] => {
+    "msg": "2000:ffff:ffff:ffff:ffff:ffff:ffff:fffb"
+}
+
+"""
+
+RETURN = """
+  data:
+    type: list
+    elements: str
+    description:
+      - Returns list with values valid for a particular query.
+"""
+
+
+@pass_environment
+def _ipmath(*args, **kwargs):
+    """Convert the given data from json to xml."""
+    keys = ["value", "amount"]
+    data = dict(zip(keys, args[1:]))
+    data.update(kwargs)
+    aav = AnsibleArgSpecValidator(
+        data=data, schema=DOCUMENTATION, name="ipmath"
+    )
+    valid, errors, updated_data = aav.validate()
+    if not valid:
+        raise AnsibleFilterError(errors)
+    return ipmath(**updated_data)
+
+
+def ipmath(value, amount):
+    try:
+        if "/" in value:
+            ip = netaddr.IPNetwork(value).ip
+        else:
+            ip = netaddr.IPAddress(value)
+    except (netaddr.AddrFormatError, ValueError):
+        msg = "You must pass a valid IP address; {0} is invalid".format(value)
+        raise AnsibleFilterError(msg)
+
+    if not isinstance(amount, int):
+        msg = (
+            "You must pass an integer for arithmetic; "
+            "{0} is not a valid integer"
+        ).format(amount)
+        raise AnsibleFilterError(msg)
+
+    return str(ip + amount)
+
+
+class FilterModule(object):
+    """IP address and network manipulation filters
+    """
+
+    filter_map = {
+        # IP addresses and networks
+        "ipmath": _ipmath
+    }
+
+    def filters(self):
+        if netaddr:
+            return self.filter_map
+        else:
+            # Need to install python's netaddr for these filters to work
```

Comment:
can we update comments lingo to be more of a dev doc than generic statements, I think that makes the public code more formal?

Reply:
ack

Reply:
ack

Reply:
added this "Need to install python's netaddr for these filters to work" in _need_netaddr function https://github.com/ansible-collections/ansible.utils/pull/115/commits/3e186211a3ef40653917acabd33cf8e1b29da737